lib/timer.js

```diff
@@ -201,26 +201,33 @@ timers.create(id, options, (err) => {
       debug('setting timer', id);
       //TODO(esatterwhite):
       // what should happen if leveldb fails.
-      if (err) console.error(err);
+      if (err) {
+        console.error(err);
+        cb(err, null);
+      }
       this.nats.publish('skyring:events', JSON.stringify({
         type: EVENT_STATUS.CREATED
       , timer: id
       , node: this[kNode]
       , created: data.created
       , payload: payload
-      }), noop);
-
-      data.timer = setTimeout(
-        transport
-      , payload.timeout - elapsed
-      , payload.callback.method
-      , payload.callback.uri
-      , payload.data
-      , id
-      , this
-      ).unref();
-      this.set( id, data );
-      cb(null, id);
+      }), (err) => {
+        if (err) {
+          console.error(err);
+          cb(err, null);
+        }
+        data.timer = setTimeout(
+          transport
+          , payload.timeout - elapsed
+          , payload.callback.method
+          , payload.callback.uri
+          , payload.data
+          , id
+          , this
+        ).unref();
+        this.set(id, data);
+        cb(null, id);
+      });
```

Comment:
Sometimes 201s were being returned before the timer was actually put into persistent storage due to the noop

Reply:
I'm not sure the two things were directly related. The the nats.publish is just for external monitoring. It doesn't have any bearing on the  actual behavior of the server.

Reply:
This still seems like a race between the callback being called and the write ack between mongo.
The timer was already being set after the leveldb write happened. All you've done here is push that down to after a nats call which is just for monitoring.

I wonder if it something with the default write concern on mongo or if there is something that can be passed to the driver...